File: bin/Data/AUI/Periodic/Sources/code_apromptwindow.cpp
// APromptWindow application source code
#include <Urho3D/AUI/AEvents.h>
#include <Urho3D/AUI/APromptWindow.h>
#include "PeriodicApp.h"

void PeriodicApp::setup_apromptwindow( AWidget *layout, AView *uiview )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleApromptwindowEvent ));
}

void PeriodicApp::HandlePromptCompleteEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace APromptComplete;

    AppLog( "APromptWindow event : the window " + eventData[P_TITLE].GetString()
            + " file was " + eventData[P_SELECTED].GetString()
            + ", the button pressed was " + eventData[P_REASON].GetString());
}

void PeriodicApp::HandleApromptwindowEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uipromptwindowcode" )
        {
            AppLog( "APromptWindow support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_apromptwindow.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uipromptwindowlayout" )
        {
            AppLog( "APromptWindow support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_apromptwindow.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() == "stringfinder" )
        {
            AppLog( "APromptWindow action : " + widget->GetId() + " was pressed ");
            AWidget *someview = (AWidget *)widget->GetView();
            APromptWindow *windowp = new  APromptWindow(context_, someview, widget->GetId());
            SubscribeToEvent( windowp, E_APROMPTCOMPLETE, URHO3D_HANDLER(PeriodicApp, HandlePromptCompleteEvent ));
            windowp->Show( "WindowTitle", "Message in window", "preset value", 0, 0, 0);
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_apromptwindow.lua
-- APromptWindow application source code


function setup_uipromptwindow( layout, uiview )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUipromptwindowEvent")
    end
end

function HandlePromptCompleteEvent(eventType, eventData)
    AppLog( "APromptWindow event : the window " .. eventData["Title"]:GetString()
            .. " file was " .. eventData["Selected"]:GetString()
            .. ", the button pressed was " .. eventData["Reason"]:GetString())
end

function HandleUipromptwindowEvent( eventType, eventData)
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
        return
    end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uipromptwindowcode" then
            AppLog( "APromptWindow support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_apromptwindow.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uipromptwindowlayout" then
            AppLog( "APromptWindow support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_apromptwindow.ui.txt", widget:GetParent())
        end
        if widget:GetId() == "stringfinder" then
            AppLog( "APromptWindow action : " .. widget:GetId() .. " was pressed ")
            local windowp = APromptWindow:new( uiview, widget:GetId())
            SubscribeToEvent(windowp, "APromptComplete", "HandlePromptCompleteEvent")
            windowp:Show( "WindowTitle", "Message in window", "preset value", 0, 0, 0)
        end
    end
end

